relayer/relays/beacon/header/header.go

```diff
@@ -319,7 +352,42 @@ func (h *Header) populateClosestCheckpoint(slot uint64) (cache.Proof, error) {
 	return checkpoint, nil
 }
 
+func (h *Header) populateCheckPointCacheWithDataFromChain(slot, checkpointSlot uint64) (uint64, error) {
+	log.WithFields(log.Fields{"calculatedCheckpointSlot": checkpointSlot}).Info("checkpoint slot not available, try with slot in next sync period instead")
+
+	checkpointSlot = h.syncer.CalculateNextCheckpointSlot(slot)
+
+	lastFinalizedHeaderState, err := h.writer.GetLastFinalizedHeaderState()
+	if err != nil {
+		return 0, fmt.Errorf("get last finalized header for the checkpoint: %w", err)
+	}
+
+	if slot > lastFinalizedHeaderState.BeaconSlot {
+		return 0, ErrBeaconHeaderNotFinalized
+	}
+
+	if checkpointSlot < lastFinalizedHeaderState.BeaconSlot {
+		historicState, err := h.findClosestCheckPoint(slot)
+		if err != nil {
+			return 0, fmt.Errorf("get history finalized header for the checkpoint: %w", err)
+		}
+		checkpointSlot = historicState.BeaconSlot
+	} else {
+		// Setting the checkpoint slot to what is the latest finalized header on-chain, since the checkpoint should
+		// not be after the latest finalized header on-chain
+		checkpointSlot = lastFinalizedHeaderState.BeaconSlot
+	}
+
+	err = h.populateFinalizedCheckpoint(checkpointSlot)
+	if err != nil {
+		return 0, fmt.Errorf("populated local cache with finalized header found on-chain: %w", err)
+	}
+
+	return 0, nil
+}
+
 func (h *Header) getHeaderUpdateBySlot(slot uint64) (scale.HeaderUpdatePayload, error) {
+	slot = slot + 1
```

Comment:
Why do we add `slot+1` here?  Is that tested with the `register_token`  smoke test still worked?

Reply:
Good catch 😄 https://github.com/Snowfork/snowbridge/pull/1156/commits/0da2b16cf053b4cbc15baab874f93c7271cab0ae